elastic_integration plugin build broken due to recent GeoIP change in ES.

### Description

`elastic_integration` plugin build and E2E (builds the plugin and runs E2E tests) BK pipelines ([build CI failure](https://buildkite.com/elastic/logstash-filter-elastic-integration-build/builds/162#018f06fa-eb5f-4247-8ac9-3237555a7514), [E2E CI failure](https://buildkite.com/elastic/logstash-filter-elastic-integration-build/builds/163#018f072d-aaf0-489c-aff1-4eff16d2533f)) are failing due to [recent GeoIP change](https://github.com/elastic/elasticsearch/pull/107377) in Elasticsearch.

```
612.3 /usr/share/plugins/plugin/src/main/java/co/elastic/logstash/filters/elasticintegration/geoip/GeoIpDatabaseAdapter.java:26: error: GeoIpDatabaseAdapter is not abstract and does not override abstract method getEnterprise(InetAddress) in GeoIpDatabase
612.3 public class GeoIpDatabaseAdapter implements GeoIpDatabase, Closeable {
612.3        ^
```

--- a/src/main/java/co/elastic/logstash/filters/elasticintegration/geoip/GeoIpDatabaseAdapter.java
+++ b/src/main/java/co/elastic/logstash/filters/elasticintegration/geoip/GeoIpDatabaseAdapter.java
@@ -11,9 +11,11 @@ import org.elasticsearch.ingest.geoip.shaded.com.maxmind.db.CHMCache;
 import org.elasticsearch.ingest.geoip.shaded.com.maxmind.db.NodeCache;
 import org.elasticsearch.ingest.geoip.shaded.com.maxmind.geoip2.DatabaseReader;
 import org.elasticsearch.ingest.geoip.shaded.com.maxmind.geoip2.model.AbstractResponse;
+import org.elasticsearch.ingest.geoip.shaded.com.maxmind.geoip2.model.AnonymousIpResponse;
 import org.elasticsearch.ingest.geoip.shaded.com.maxmind.geoip2.model.AsnResponse;
 import org.elasticsearch.ingest.geoip.shaded.com.maxmind.geoip2.model.CityResponse;
 import org.elasticsearch.ingest.geoip.shaded.com.maxmind.geoip2.model.CountryResponse;
+import org.elasticsearch.ingest.geoip.shaded.com.maxmind.geoip2.model.EnterpriseResponse;
 
 import java.io.Closeable;
 import java.io.File;
@@ -50,6 +52,16 @@ public class GeoIpDatabaseAdapter implements GeoIpDatabase, Closeable {
     @Override
     public AsnResponse getAsn(InetAddress inetAddress) {
         return getResponse(inetAddress, this.databaseReader::tryAsn);
+    }
+
+    /* @Override // neither available nor reachable until Elasticsearch 8.14 */
+    public AnonymousIpResponse getAnonymousIp(InetAddress ipAddress) {
+        return getResponse(ipAddress, this.databaseReader::tryAnonymousIp);
+    }
+
+    /* @Override // neither available nor reachable until Elasticsearch 8.14 */
+    public EnterpriseResponse getEnterprise(InetAddress ipAddress) {
+        return getResponse(ipAddress, this.databaseReader::tryEnterprise);
     }
 
     private <T extends AbstractResponse> T getResponse(final InetAddress inetAddress, MaxmindTryLookup<T> resolver) {

